reconcile/utils/promotion_state.py

```diff
@@ -95,7 +95,9 @@ def get_promotion_data(
         if use_cache and path_v2 in self._promotion_data_cache:
             return self._promotion_data_cache[path_v2]
 
-        data = self._state.get(path_v2)
+        data = self._state.get(path_v2, None)
+        if not data:
+            return None
```

Comment:
Should be caught by https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/test/utils/test_promotion_state.py#L66

Reply:
Apologies, that test is for cached commit shas ... We dont have a test for non-existing key without caching yet. Im happy to take a look tomorrow.